src/java.base/share/classes/java/lang/runtime/ExactConversionsSupport.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package java.lang.runtime;
+
+/**
+ * A casting conversion is considered to be exact if, after applying the conversion
+ * to the value, no exception was raised or loss of information has occured;
+ * otherwise it is considered inexact.
+ * The methods in this class provide the run-time support for primitive conversions exactness checks.
+ * These methods may be used, for example, by Java compiler implementations to implement checks
+ * for instanceof and pattern matching runtime implementations.
+ * See JLS section 5.5.1 for more information on exact casting conversions.
+ *
+ * @jls 5.5.1 Exact Casting Conversions
+ * @jls 15.20.2 The instanceof Operator
+ *
+ * @since 22
+ */
+public final class ExactConversionsSupport {
+
+    private ExactConversionsSupport() { }
+
+     /** Exactness method from int to byte
+     *
+     * @param n value
+     * @return  true if the passed value can be converted exactly to the target type
+     *
+     * */
+    public static boolean intToByteExact(int n)      {return n == (int)(byte)n;}
```

Comment:
These ones seem more _conversions_ than _tests_.
To me, `intToByteExact(int)` looks more like a request to convert an `int` to a `byte`, rather than a check to see whether the argument lies in the `byte` range.
Pure methods returning `boolean` are often named like `isIntToByteExact(int)` or the like (`is*()`, `has*()`, etc.)

I know: choosing good, speaking, and evocative names is one of the hardest problem in computer science.

Reply:
Good recommendation. Names updated!